cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -102,11 +80,32 @@ public void init(String[] args) {
       }
     }
 
-    int brokerId = generateBrokerId(address);
-    LOG.info(String.format("Initializing server with broker id %d", brokerId));
+    kafkaProperties = generateKafkaConfig(cConf);
 
-    kafkaProperties = generateKafkaConfig(brokerId, zkConnectStr, hostname, port, numPartitions,
-                                          replicationFactor, logDir);
+    Preconditions.checkState(Integer.parseInt(
+      kafkaProperties.getProperty(KafkaConstants.ConfigKeys.NUM_PARTITIONS_CONFIG)) > 0,
+                             "Num partitions should be greater than zero.");
+    int port = Integer.getInteger(kafkaProperties.getProperty(KafkaConstants.ConfigKeys.PORT_CONFIG), -1);
+    Preconditions.checkState(port > 0, "Port number is invalid.");
+
+    String hostname = (String)kafkaProperties.get("host.name") ;
+    InetAddress address = Networks.resolve(hostname, null);
+    if (hostname != null) {
+      if (address != null && address.isAnyLocalAddress()) {
+        kafkaProperties.remove("host.name");
+      } else {
+        hostname = address.getCanonicalHostName();
+        kafkaProperties.setProperty("host.name", hostname);
+      }
+    }
+    if (kafkaProperties.get("broker.id") == null) {
+      int brokerId = generateBrokerId(address);
+      LOG.info(String.format("Initializing server with broker id %d", brokerId));
+      kafkaProperties.setProperty("broker.id", Integer.toString(brokerId));
+    }
+    if (kafkaProperties.getProperty("zookeeper.connect") != null) {
```

Comment:
hm. But isn't the point here to configure the proper hostname value? Since if the user sets 'kafak.server.host.name" as `0.0.0.0`, `kafkaProperties.get("host.name")` will indeed be not `null`. It is that we now need to unset it to allow for listening to all interfaces.